### ISSUE TITLE
Add a `mempool_transactions()` RPC method for listening to changes in the mempool

### ISSUE THREAD

**Issue by arya2** [ZcashFoundation/zebra#8620]
## Motivation

We want to notify RPC clients when there has been a change to the mempool with information about which transactions were added or removed.

## Design

- Add a `MempoolChange` enum with `Added`, `Invalidated`, and `Mined` variants
- Add a channel for the mempool to broadcast changes to its verified tx set
- Send a `MempoolChange` when a transaction is inserted, invalidated by a chain tip change, or mined onto the best chain
- Add a receiver for the new channel to the indexer RPC server (or the regular RPC server now that it uses `jsonrpsee`)
- Add an RPC method that notifies clients of mempool changes
  - Send any existing mempool transactions when the method is called as `Added` changes
  
<details>
<summary>Outdated</summary>
[add a `MempoolChange` type with an `async changed()` method (like `ChainTip`).](https://github.com/ZcashFoundation/zebra/blob/main/zebra-rpc/src/methods/get_block_template_rpcs.rs#L690C29-L690C42)

This would require adding a mempool change channel that could help clear [this TODO](https://github.com/ZcashFoundation/zebra/blob/main/zebra-rpc/src/methods/get_block_template_rpcs.rs#L741-L743).
</details>

**Comment by arya2** [comment]
@idky137 Is this something Zaino would use, or is there already an implementation that uses polling?

**Comment by idky137** [comment]
Yes we currently Poll `get_raw_mempool` and filter known "mempool transactions" but this would be more efficient.

Would this method start by giving all of the current "mempool transactions" (or could this be an option)? If not we would have to call `get_raw_mempool` followed by `mempool_transactions`, and I think we could end up missing mempool changes in this gap.

This would definitely be useful though!

**Comment by arya2** [comment]
> Would this method start by giving all of the current "mempool transactions" (or could this be an option)?

I've updated the design to include sending existing transactions, let us know if you have any notes on the design, or if there's any other information it should be returning.
